Simplenote/src/main/java/com/automattic/simplenote/NoteEditorFragment.java

```diff
@@ -355,7 +355,8 @@ public void bindView(View view, Context context, Cursor cursor) {
 
             @Override
             public CharSequence convertToString(Cursor cursor) {
-                return cursor.getString(cursor.getColumnIndex(Tag.NAME_PROPERTY));
+                int val = cursor.getColumnIndex(Tag.NAME_PROPERTY);
+                return cursor.getString(val);
```

Comment:
@jkmassel In my opinion, this doesn't address the lint error, it just tricks it so the error is not reported anymore 😅 If you replace `getColumnIndex` with `getColumnIndexOrThrow`, the lint error should be gone and it'd work as we expect it to.

For context, the error being reported is that [getColumnIndex](https://developer.android.com/reference/android/database/Cursor#getColumnIndex(java.lang.String)) would return `-1` if the column doesn't exist. If we use [getColumnIndexOrThrow](https://developer.android.com/reference/android/database/Cursor#getColumnIndexOrThrow(java.lang.String)), we'd still get a crash, but the reporting would be more useful because it'd point to the actual issue instead of a bad input when we use the result of `getColumnIndex`.

Also worth noting that we have fixed this issue by replacing `getColumnIndex` with `getColumnIndexOrThrow` in other projects. [Here is an example of one.](https://github.com/tumblr/PermissMe/pull/2/files#diff-846b9c15d65697055981fd6c02c8e4d9a30eb36bf5d07df29ea24c0d16392875R38)

Reply:
Good call – this code is very old, so I wasn't sure if we were really saving anything tbh.

That said, if we can crash instead of propagating invalid values, that's of course better!

Adjusted to use `getColumnIndexOrThrow` everywhere in fa368a6.